### ISSUE TITLE
Non-Ascii  characters Supports

### ISSUE THREAD

**Issue by CNBroderick** [a-schild/nextcloud-java-api#19]
Url cations non-ascii will have exception

Flow that titles, I have a lasted next cloud 14 server in there. And I have a game mods will to upload timely. 
Contains Non ASCii characters url will have a exception when upload. And I have a resolved method:
```Java
    private String getEncodeUrl(String s) {
        try {
            return URLEncoder.encode(s, "utf-8");
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
        }
        return "";
    }
```

**Comment by tobiasKaminsky** [comment]
Question 1: yes, you have to encode it. On android we are using Uri.encode:

> Encodes characters in the given string as '%'-escaped octets using the UTF-8 scheme. Leaves letters ("A-Z", "a-z"), numbers ("0-9"), and unreserved characters ("_-!.~'()*") intact. Encodes all other characters with the exception of those specified in the allow argument

This should be in library so that it is in one central place and need not to be done in every application.

**Comment by a-schild** [comment]
@CNBroderick can you please move the second question to it's own issue?
It will be simpler to track and work on them separately, since they have nothing to do with each other

**Comment by a-schild** [comment]
@CNBroderick At which place in the APIU do you need to encode it manually?
The whole file/folder creation, deletion up+download should work as is.
Also the sharing API seems to work ok for me

**Comment by a-schild** [comment]
Is implemented in the 11.0.3 release.
No need to any longer encode things on your side
